browser/src/Services/FileSystemWatcher/index.ts

```diff
@@ -0,0 +1,99 @@
+import * as chokidar from "chokidar"
+import { Stats } from "fs"
+import { Event, IEvent } from "oni-types"
+
+import * as Workspace from "./../Workspace"
+
+export type Targets = string | string[]
+
+interface IFSOptions {
+    options?: chokidar.WatchOptions
+    target?: Targets
+}
+
+interface IFileChangeEvent {
+    path: string
+}
+
+interface IStatsChangeEvent {
+    path: string
+    stats: Stats
+}
+
+export class FileSystemWatcher {
+    private _watcher: chokidar.FSWatcher
+    private _workspace: Workspace.Workspace
+    private _activeWorkspace: string
+
+    private _onAdd = new Event<IFileChangeEvent>()
+    private _onAddDir = new Event<IStatsChangeEvent>()
+    private _onMove = new Event<IFileChangeEvent>()
+    private _onChange = new Event<IFileChangeEvent>()
+    private _defaultOptions = { ignored: "**/node_modules" }
+
+    constructor(watch: IFSOptions = {}) {
+        this._workspace = Workspace.getInstance()
+        this._activeWorkspace = this._workspace.activeWorkspace
```

Comment:
I'm wondering - would it make sense to hook the `onDirectoryChanged` event on `Workspace`? In that case, we'd probably want to remove the existing listeners from the old directory, and start listening to the 'new' directory.